packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -1,37 +1,165 @@
+        self.samehost = samehost
+        self.vipnet = vipnet
+        self.proxynet = proxynet
+
+    def __str__(self):
+        return ("VipTest(vip={},vipaddr={},samehost={},"
+                "vipnet={},proxynet={})").format(self.vip, self.vipaddr, self.samehost, self.vipnet, self.proxynet)
+
+    def log(self, s, lvl=logging.DEBUG):
+        m = 'VIP_TEST {} {}'.format(s, self)
+        log.log(lvl, m)
+
+
+def vip_test(dcos_api_session, r):
+    r.log('START')
+    agents = list(dcos_api_session.slaves)
+    # make sure we can reproduce
+    random.seed(r.vip)
+    random.shuffle(agents)
+    host1 = agents[0]
+    host2 = agents[0]
+    if not r.samehost:
+        host2 = agents[1]
+    log.debug('host1 is is: {}'.format(host1))
+    log.debug('host2 is is: {}'.format(host2))
+    origin_app, app_uuid = get_test_app_in_docker()
+    origin_app['container']['docker']['portMappings'][0]['labels'] = {'VIP_0': r.vip}
+    origin_app['container']['docker']['network'] = r.vipnet
+    if r.vipnet == 'USER':
+        origin_app['ipAddress'] = {'networkName': 'dcos'}
+    if r.vipnet == 'HOST':
+        origin_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                            '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
+        origin_app['container']['docker']['portMappings'][0]['hostPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['containerPort'] = 0
+        origin_app['container']['docker']['portMappings'][0]['labels'] = {}
+        origin_app['portDefinitions'] = [{'labels': {'VIP_0': r.vip}}]
+
+    origin_app['constraints'] = [['hostname', 'CLUSTER', host1]]
+
+    proxy_app = get_test_app_in_docker()[0]
+    proxy_app['container']['docker']['network'] = r.proxynet
+    proxy_app['constraints'] = [['hostname', 'CLUSTER', host2]]
+    if r.proxynet == 'USER':
+        proxy_app['ipAddress'] = {'networkName': 'dcos'}
+    if r.proxynet == 'HOST':
+        proxy_app['cmd'] = '/opt/mesosphere/bin/dcos-shell python '\
+                           '/opt/mesosphere/active/dcos-integration-test/util/python_test_server.py $PORT0'
+        proxy_app['container']['docker']['portMappings'][0]['hostPort'] = 0
+        proxy_app['container']['docker']['portMappings'][0]['containerPort'] = 0
+
+    returned_uuid = None
+
+    with contextlib.ExitStack() as stack:
+        stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(origin_app, timeout=timeout))
+        sp = stack.enter_context(dcos_api_session.marathon.deploy_and_cleanup(proxy_app, timeout=timeout))
+        cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://{}/test_uuid'.format(r.vipaddr)
+        returned_uuid = ensure_routable(cmd, sp)
+        log.debug('returned_uuid is: {}'.format(returned_uuid))
+    assert returned_uuid is not None
+    assert returned_uuid['test_uuid'] == app_uuid
+    r.log('PASSED')
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
```

Comment:
Skipping a fixture doesnt make too much sense, you want this decorator around test_vip I think